frontend/containers/settings-security/2fa/component.tsx

```diff
@@ -0,0 +1,70 @@
+import { FC, useEffect } from 'react';
+
+import { useForm } from 'react-hook-form';
+import { FormattedMessage } from 'react-intl';
+
+import useMe from 'hooks/me';
+
+import Label from 'components/forms/label';
+import Switch from 'components/forms/switch';
+
+import { useUpdateUser } from 'services/users/userService';
+
+import { Settings2FAProps } from '.';
+
+const Settings2FA: FC<Settings2FAProps> = () => {
+  const { user, isLoading } = useMe();
+  const { register, setValue, handleSubmit } = useForm<{ otp_required_for_login: boolean }>();
+  const updateUser = useUpdateUser();
+
+  useEffect(() => {
+    if (!!user) {
+      setValue('otp_required_for_login', user.otp_required_for_login);
+    }
+  }, [setValue, user]);
+
+  const onSubmit = ({ otp_required_for_login }: { otp_required_for_login: boolean }) => {
+    if (!!user) {
+      // The only value extracted from this form is the otp_required_for_login, the other values come from the current user.
+      const { first_name, last_name } = user;
+      updateUser.mutate({ first_name, last_name, otp_required_for_login });
```

Comment:
Do you really need to send `first_name` and `last_name`? Have you tried only sending `otp_required_for_login`?